File: 0179-largest-number/0179-largest-number.cpp
class Solution {
public:
    bool myCompetitor(string &s1,string &s2){
        if(s1+s2 > s2+s1){
            return true;
        }
        return false;
    }
    string largestNumber(vector<int>& nums) {
        vector<string> st;
        st.push_back(to_string(nums[0]));

        for(int i=1;i<nums.size();i++){
            st.push_back(to_string(nums[i]));

            int j = st.size()-1;
            while(j>0){
                if(myCompetitor(st[j],st[j-1])){
                    swap(st[j],st[j-1]);
                }
                else{
                    break;
                }
                j--;
            }
        }
        string result ="";
        for(int i=0;i<st.size();i++){
            result +=st[i];
        }
        return result;
    }
   
};

File: 0179-largest-number/0179-largest-number.py
def to_swap(num1,num2):
    if(int(str(num2)+str(num1)) < int(str(num1)+str(num2))):
        return True
    return False
class Solution(object):
    def largestNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: str
        """
        l1 = []
        l1.append(nums[0])
        i=1
        while(i<len(nums)):
            l1.append(nums[i])
            j = len(l1)-1
            while(j>0):
                if(to_swap(l1[j],l1[j-1])):
                    l1[j],l1[j-1] = l1[j-1],l1[j]
                j -=1
            i +=1
        
        ans = ""
        for ch in l1:
            ans +=str(ch)
        if(int(ans) == 0):
            return '0'
        return ans

        